### PR TITLE
Testdata for PR #17978

### PR DESCRIPTION
**Pull request by sl-sergei**
Data for PR https://github.com/opencv/opencv/pull/17978

### CODE REVIEW

**Comment by alalek** [comment]
Please rebase commit to resolve conflicts.

File: testdata/dnn/onnx/generate_onnx_models.py [hunk 1]
```diff
@@ -877,6 +877,51 @@ def forward(self, x, kernel, bias):
 model = ConvBias()
 save_data_and_model_multy_inputs("conv_variable_wb", model, x, kernel, bias)
 
+<<<<<<< HEAD
```
Comment (l-bat) [inline testdata/dnn/onnx/generate_onnx_models.py:880]: Please remove this line

File: testdata/dnn/onnx/generate_onnx_models.py [hunk 2]
```diff
@@ -877,6 +877,51 @@ def forward(self, x, kernel, bias):
...
+        const_projected = self.linear_for_const(const)
+        const_projected = const_projected.expand(5, self.interm_dim)
+        sum_tanh = torch.tanh(const_projected + x_projected) 
+        sum_tanh = sum_tanh.reshape(-1, self.interm_dim)
+        sum_tanh_projected = self.second_linear(sum_tanh)
+        sum_tanh_projected = sum_tanh_projected.reshape(1, 5)
+        after_softmax = F.softmax(sum_tanh_projected, dim=1)     
+        return torch.matmul(after_softmax, x)
+
+x = Variable(torch.rand([1, 5, 10]))
```
Comment (l-bat) [inline testdata/dnn/onnx/generate_onnx_models.py:921]: Could you create test data with smaller sizes?

File: testdata/dnn/onnx/generate_onnx_models.py (hunk 1 above)
Comment (sl-sergei) [reply testdata/dnn/onnx/generate_onnx_models.py]: Sorry, my development tools produce these strange bugs (not shown in IDE)

File: testdata/dnn/onnx/generate_onnx_models.py (hunk 2 above)
Comment (sl-sergei) [reply testdata/dnn/onnx/generate_onnx_models.py]: Done

File: testdata/dnn/onnx/generate_onnx_models.py [hunk 3]
```diff
@@ -887,3 +887,47 @@ def forward(self, x, kernel, bias):
...
+        x_projected = self.first_linear(x)
+        const = torch.zeros(1, self.interm_dim)
+        const_projected = self.linear_for_const(const)
+        const_projected = const_projected.expand(2, self.interm_dim)
+        sum_tanh = torch.tanh(const_projected + x_projected) 
+        sum_tanh = sum_tanh.reshape(-1, self.interm_dim)
+        sum_tanh_projected = self.second_linear(sum_tanh)
+        sum_tanh_projected = sum_tanh_projected.reshape(1, 2)
+        after_softmax = F.softmax(sum_tanh_projected, dim=1)     
+        return torch.matmul(after_softmax, x)
```
Comment (l-bat) [inline testdata/dnn/onnx/generate_onnx_models.py:929]: Do we really need such a large subgraph? Maybe you can cover your changes in the onnx importer using just one matrix multiplication?

Comment (sl-sergei) [reply testdata/dnn/onnx/generate_onnx_models.py:929]: Unfortunately, it does not reproduce that way

**Comment by alalek** [comment]
Please do not use **merge commits** in PRs to resolve conflicts.
Use `git rebase` instead.

There are several artifacts during PR merge which requires extra changes:
```
* Add new testdata for Matmul and Expand operations

* remove artifacts and reduce data size

* Added testdata for ONNX Pow op

* Add regression data for NaN output with Andreff's Hand-Eye calibration method.

Co-authored-by: Liubov Batanina <piccione-mail@yandex.ru>
Co-authored-by: catree <catree.catreus@outlook.com>
```

**Pull request merged**